wordpress-mobile/WordPress-Android
# Unified Dashboard: Menu UI cleanup

Parent #19270 

This PR applies the following the the Menu view

- Updates general spacing/styling of items
- Adjusts the way the CustomXMLWidgetView is composed to reserve resources
- Extracts the composables out of the activity, but remains in the same fil
- Adds RTL support
- Create MenuItemState and swap that in for MySiteCardAndItem
- Add base unit tests for MenuViewModel

**To test:**
- Install the app
- Login and select a site
- Navigate to Quick Links Card > More
- ✅ Verify the view loads correctly and look good
- ✅ Verify the view looks the same in dark and light mode
- Change the app language to a RTL language (hebrew works great)
- ✅ Verify the view looks is shown RTL
- Go through the remainder of the UI checklist and verify all works as expected

## Regression Notes
1. Potential unintended areas of impact
The menu looks poorly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

UI Changes testing checklist:
